File: app/src/main/java/ru/tsu/quizium/RecyclerViewItem.java
package ru.tsu.quizium;

public class RecyclerViewItem {

    private final Quiz quiz;
    private final Category category;
    private final int typeCode;

    private final RecyclerViewDataTypeCode type = new RecyclerViewDataTypeCode();

    RecyclerViewItem(Quiz quiz){
        this.quiz = quiz;
        this.category = null;
        this.typeCode = type.TYPE_QUIZ;
    }
    RecyclerViewItem(Category category){
        this.quiz = null;
        this.category = category;
        this.typeCode = type.TYPE_CATEGORY;
    }

    public int getTypeCode(){
        return typeCode;
    }

    public Quiz getQuiz(){
        return quiz;
    }

    public Category getCategory() {
        return category;
    }
}

File: app/src/main/java/ru/tsu/quizium/RecyclerViewItem.kt
package ru.tsu.quizium

class RecyclerViewItem {
    val quiz: Quiz?
    val category: Category?
    val typeCode: Int

    internal constructor(quiz: Quiz?) {
        this.quiz = quiz
        category = null
        typeCode = AppConst.TYPE_QUIZ
    }

    internal constructor(category: Category?) {
        quiz = null
        this.category = category
        typeCode = AppConst.TYPE_CATEGORY
    }

}